Framework/Algorithms/src/SpecularReflectionAlgorithm.cpp

```diff
@@ -213,9 +213,8 @@ double SpecularReflectionAlgorithm::calculateTwoTheta() const {
   IComponent_const_sptr detector =
       this->getDetectorComponent(inWS, analysisMode == pointDetectorAnalysis);
 
-  IComponent_const_sptr sample = this->getSurfaceSampleComponent(instrument);
-
-  const V3D detSample = detector->getPos() - sample->getPos();
+  const V3D detSample =
+      detector->getPos() - inWS->spectrumInfo().samplePosition();
```

Comment:
I was a bit too quick to comment on this previously, sorry about that! If I understood correctly, there is a `SampleComponentName` property which enables a user to name a specific component as the sample. This property is read in `SpecularReflectionAlgorithm::getSurfaceSampleComponent()`, on the now-deleted line above. If `SampleComponentName` is not specified, `getSurfaceSampleComponent()` makes an educated guess that the instrument has a component called 'some-surface-holder'. I don't know how many instruments actually have a component with that name, but I do know that many instruments have a component specifically tagged as Sample in the IDF. So, what I should have proposed here in the first place, is to
- check if `SampleComponentName` is given
- if yes, find the component's position
- if no, use `SpectrumInfo::samplePosition()` because it is usually better guess than 'some-surface-holder'
In any case, this change should be reverted as it effectively disables `SampleComponentName`.

Reply:
Oops, I hadn't spotted this either! That's a good suggestion, and `SpecularReflectionPositionCorrect` should probably use the same method too. I'll create a separate issue for this and revert this change for now.